src/cloudWatchLogs/activation.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { LogStreamDocumentProvider } from './document/logStreamDocumentProvider'
+import { CLOUDWATCH_LOGS_SCHEME } from './constants'
+import { LogStreamRegistry } from './registry/logStreamRegistry'
+import { viewLogStream } from './commands/viewLogStream'
+import { LogGroupNode } from './explorer/logGroupNode'
+
+export async function activate(context: vscode.ExtensionContext): Promise<void> {
+    const registry = new LogStreamRegistry()
+
+    const logStreamProvider = new LogStreamDocumentProvider(registry)
+
+    context.subscriptions.push(
+        vscode.workspace.registerTextDocumentContentProvider(CLOUDWATCH_LOGS_SCHEME, logStreamProvider)
+    )
+
+    // these actions are typically in the explorer activation but this requires knowledge of the registry.
```

Comment:
the comment is vague and is unlikely to be meaningful outside of the moment it was written...

Reply:
Retooled it. Thinking it's a nice-to-have since it's unconventional for Explorer actions.